net/mdns-repeater/src/opnsense/mvc/app/views/OPNsense/MDNSRepeater/index.volt

```diff
@@ -31,29 +31,104 @@ POSSIBILITY OF SUCH DAMAGE.
         updateServiceStatusUI(data['result']);
     });
 
     // link save button to API set action
-    $("#saveAct").click(function(){
-        saveFormToEndpoint(url="/api/mdnsrepeater/settings/set", formid='general',callback_ok=function(){
-            $("#saveAct_progress").addClass("fa fa-spinner fa-pulse");
-            ajaxCall(url="/api/mdnsrepeater/service/restart", sendData={}, callback=function(data,status) {
-                ajaxCall(url="/api/mdnsrepeater/service/status", sendData={}, callback=function(data,status) {
-                    updateServiceStatusUI(data['result']);
-                });
-                $("#saveAct_progress").removeClass("fa fa-spinner fa-pulse");
-            });
-        });
-    });
+    //~ $("#saveAct").click(function(){
+        //~ saveFormToEndpoint(url="/api/mdnsrepeater/settings/set", formid='general',callback_ok=function(){
+            //~ $("#saveAct_progress").addClass("fa fa-spinner fa-pulse");
+            //~ ajaxCall(url="/api/mdnsrepeater/service/restart", sendData={}, callback=function(data,status) {
+                //~ ajaxCall(url="/api/mdnsrepeater/service/status", sendData={}, callback=function(data,status) {
+                    //~ updateServiceStatusUI(data['result']);
+                //~ });
+                //~ $("#saveAct_progress").removeClass("fa fa-spinner fa-pulse");
+            //~ });
+        //~ });
+    //~ });
 });
 
 </script>
 
+<script type="text/javascript">
+
+    /**
+     * prepare for checking scan status
+     */
+    //~ function scanStatusPrepare(rerun) {
+        //~ if ($rerun = false) {
+            //~ $('#scan_status').hide();
+            //~ $('#scanlist').show();
+        //~ }
+        //~ $("#checkscan_progress").addClass("fa fa-spinner fa-pulse");
+        //~ $('#scanstatus').html("{{ lang._('Checking... (may take up to 30 seconds)') }}");
+    //~ }
+
+    /**
+     * retrieve scan status from backend
+     */
+    function scanStatus() {
+        // scan UI
+        //~ scanStatusPrepare(false);
+
+        // request status
+        ajaxGet('/api/arpscanner/status',{},function(data,status){
+            $("#checkscan_progress").removeClass("fa fa-spinner fa-pulse");
+            $('#scanstatus').html(data['status_msg']);
+
+            if (data['status'] == "1") {
+
+                //~ run audit
+                
+            }
+        });
+    }
+
+    /**
+     * perform upgrade, install poller to scan status
+     */
+    function stopScan() {
+        $('#scanlist').hide();
+        $('#scan_status').show();
+        $('#scantab > a').tab('show');
+        $('#scanstatus').html("{{ lang._('Killing...') }}");
+        $("#audit").attr("style","display:none");
+        maj_suffix = '';
+        if ($.upgrade_action == 'maj') {
+            maj_suffix = '_maj';
+        }
+        $("#upgrade" + maj_suffix).attr("style","");
+        $("#upgrade_progress" + maj_suffix).addClass("fa fa-spinner fa-pulse");
```

Comment:
???
something copied from the firmware page?

Reply:
I followed your suggestion about copy an working example from firmware. I will not use all this, I just leave it waiting for the first revision. I think that We could do something more about realtime auditing. I develop also in django channels, have you ever thought to migrate from phalcon to django framework? :)

Reply:
Some users have asked that question at the beginning (15.x series). The answer is that with Phalcon, lots of code can be reused and no legacy code needs to be dropped so development can continue using the new code while keeping support for the old code to slowly migrate everything. The long term goal is to have everything in MVC and run the web server and PHP process as a user (not as root) and privileged actions using configd. Phalcon is also very fast so there is no reason to migrate away from it.

Reply:
@fichtner Please correct me if I am wrong.

Reply:
@fabianfrz what you say sound like the correct choice. If I can give you support, probably, I could in python, in the configd layer. It doesn't sound to me as an evolute software, If help is needed you can always ask

Reply:
#upgrade will not exist anymore on that page. Do I have to remove that commented lines?

Reply:
yes